pdg/src/builder.rs

```diff
@@ -192,15 +208,19 @@ pub fn add_node(
 
     let node = Node {
         function: Func(dest_fn),
-        block: basic_block_idx.clone().into(),
-        index: statement_idx.clone().into(),
+        block: basic_block_idx.into(),
+        index: statement_idx.into(),
         kind: node_kind,
-        source: source.and_then(|p| get_parent_object(&event.kind, p)).map(|n| n.1),
+        source: source
+            .and_then(|p| event.kind.parent(p))
+            .map(|(_, nid)| nid),
```

Comment:
just FYI this is the place to tweak to write a test against, to ensure that all Nodes have source GraphId's (here ignored) that point to the graph whose node list they are a part of.

Reply:
Got it.  I'll come back to check this once the tests are in place.